### ISSUE TITLE
Fine Tuning non-Flair Language Models

### ISSUE THREAD

**Issue by BernierCR** [flairNLP/flair#1001]
Is this possible to do within flair, fine tune a lm such as WordEmbeddings('en-twitter')? 

I get that the point of Flair is to use Flair embeddings, but all these other kinds are supported to be used at least. Flair embeddings take a long time to build from scratch. WordEmbeddings would train faster right? The idea would be to see how much custom embeddings would change the models accuracy at all. Do a fast way first, and if it works well, that justifies trying a longer process. 

I can fine tune outside of Flair I guess, but if it's close to being able to do so inside Flair, I might just fix it so it can do that. 

Thanks

**Comment by Hellisotherpeople** [comment]
I want this really badly. Please implement it!

**Comment by alanakbik** [comment]
Hello @BernierCR yes that's a good point and something we should add. For standard word embeddings, we see two ways to fine-tune:
(a) the classical way in which word embeddings are used to initialize a one-hot embedding layer - this allows words to be treated separately in fine-tuning but limits the vocabulary to words in the training corpus. 
(b) a reprojection layer on top of word embeddings - this learns a (potentially non-linear) tranformation of the word embedding space, so fine-tuning is not on a per-word basis but affects all words in the embedding vocabulary in the same way. 

The second way is already implicitly implemented for all embeddings as part of the downstream tasks, for instance the `SequenceTagger`, by setting `reproject=True` (which is the default).

The first is currently only implemented for `FlairEmbeddings` - however, given the power of Flair embeddings, allowing fine-tuning almost always leads to bad overfitting and to much worse test performance, so we haven't yet found a good use case when it makes sense. But I think we can definitely look into adding fine-tuning for word embeddings and maybe the transformer based embeddings if that's possible.

**Comment by junhua** [comment]
so setting reproject=True, how to export the fine-tuned embedding at the end of the training? @alanakbik

**Comment by alanakbik** [comment]
@junhua the SequenceTagger currently does not expose the intermediary layers, so that's currently not possible. But since a few people have been asking about this, perhaps we can find a way to add this functionality.
